workflows/pr_gate/Jenkinsfile

```diff
@@ -0,0 +1,99 @@
+            "PUBLIC_JENKINS_URL=${env.PUBLIC_JENKINS_URL}",
+            "TEST_STACK=${env.TEST_STACK}",
+            "TESTS=${env.TESTS}",
+            "KEEP_FAILURE_ENV=${env.KEEP_FAILURE_ENV}",
+            "KEEP_MINUTES=${env.KEEP_MINUTES}",
+            "KEEP_DOCKER_ON_FAILURE=${env.KEEP_DOCKER_ON_FAILURE}"
+        ]){
+            deleteDir() 
+            def manifest = new pipeline.common.Manifest()
+            def share_method = new pipeline.common.ShareMethod()
+            String manifest_path = "manifest_$PR_ID"
+            String library_dir = "on-build-config"
+            def manifest_dict = null
+            
+            try{
+                stage("Parse Pull Request"){
+                    share_method.checkoutOnBuildConfig(library_dir)
+                    manifest_path = manifest.generateManifestFromPR(PR_URL, PR_TARGET_BRANCH, manifest_path, library_dir, "JENKINSRHD_GITHUB_TOKEN")
+                    archiveArtifacts "$manifest_path"
+                    manifest_dict = manifest.stashManifest("manifest", manifest_path)
+                }
+                stage("Unit Test"){
+                    ArrayList<String> used_resources = []
+                    def test_repos = manifest.getReposNeedTest(manifest_path, library_dir)
+                    def unit_test = new pipeline.rackhd.source_code.UnitTest()
+                    def unit_test_branches = [:]
+                    for(int i=0; i<test_repos.size; i++){
+                        String repo_name = test_repos.get(i)
+                        unit_test_branches["${repo_name}"] = {
+                            if(repo_name == "on-http" || repo_name == "on-core"){
+                                unit_test.runTestWithSudo(repo_name, manifest_dict, used_resources)
+                            }else{
+                                unit_test.runTestWithoutSudo(repo_name, manifest_dict, used_resources)
+                            }
+                        }
+                    }
+                    if(test_repos.size()>0){
+                        parallel unit_test_branches
+                    }
+                }
+                stage("Function Test"){
+                    def function_test = new pipeline.rackhd.source_code.FunctionTest()
+                    def function_test_branches = [:]
+                    ArrayList<String> used_resources = []
+                    List tests = Arrays.asList(TESTS.split(','))
+                    int sleep_mins = Integer.valueOf(KEEP_MINUTES)
+                    boolean keep_docker_on_failure = KEEP_DOCKER_ON_FAILURE=='true'
+                    boolean keep_failure_env = KEEP_FAILURE_ENV=='true'
+                    for(int i=0; i<tests.size(); i++){
+                        String test_name = tests[i]
+                        function_test_branches[test_name] = {
+                             withCredentials([
+                                 usernamePassword(credentialsId: 'ff7ab8d2-e678-41ef-a46b-dd0e780030e1',
```

Comment:
because the `ff7ab8d2-e678-41ef-a46b-dd0e780030e1' is hard to understand, can you add a comment: "vmslave sudo cred"?

Reply:
Thanks, fixed!